processor/transform/images.ts

```diff
@@ -0,0 +1,32 @@
+import { visit } from 'unist-util-visit';
+
+import { NodeTypes } from '../../enums';
+
+const imageTransformer = () => {
+  return (tree: any) => {
+    visit(tree, 'image', (node, _, parent) => {
+      // check if inline or already transformed
+      if (parent.type !== 'paragraph' || parent.children.length > 1 || node.data?.hName === 'image') return;
+      const newNode = {
+        type: NodeTypes.image,
+        data: {
+          hProperties: { 
+            title: node.title,
+            src: node.url,
+            alt: node.alt,
+            align: node.align,
+            border: node.border,
+            width: node.width,
+            caption: node.caption,
+            lazy: node.lazy, 
+          },
+          hName: 'image',
+        },
+        position: node.position,
+      };
+      parent = newNode;
```

Comment:
How does this work?!?! I would think it's only re-assigning the parameter?

Reply:
![2Gb](https://github.com/readmeio/markdown/assets/20567510/53d28f25-051b-4021-8f80-65ef6972b71d)

Reply:
actually idk either--i stubbed that out in the beginning thinking i'd change it later but i didn't...need to?